engine/src/main/java/io/zeebe/engine/processing/job/JobCompleteProcessor.java

```diff
@@ -73,9 +67,26 @@ public void afterAccept(
 
         // TODO (#6172) send out COMPLETE_ELEMENT command when service task processor is registered
         // for COMPLETE_ELEMENT commands; switch out for command writer
-        stateWriter.appendFollowUpEvent(
-            serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        if (MigratedStreamProcessors.isMigrated(BpmnElementType.SERVICE_TASK)) {
+          stateWriter.appendFollowUpEvent(
+              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        } else {
+          eventWriter.appendFollowUpEvent(
+              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        }
```

Comment:
I think it is enough to just have `eventWriter.appendFollowUpEvent(
              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);` we should migrate that if we migrate service task.

Reply:
I would leave it, this allows for easier migration of the service task, and this feature toggle will be removed in #6202